spark/src/main/java/org/opensearch/sql/spark/dispatcher/RefreshQueryHandler.java

```diff
@@ -59,6 +60,8 @@ public String cancelJob(AsyncQueryJobMetadata asyncQueryJobMetadata) {
   @Override
   public DispatchQueryResponse submit(
       DispatchQueryRequest dispatchQueryRequest, DispatchQueryContext context) {
+    leaseManager.borrow(new LeaseRequest(JobType.BATCH, dispatchQueryRequest.getDatasource()));
```

Comment:
Does this warrant any change in documentation?

Reply:
We're moving this from parent class? The new create index job doesn't need this?

Reply:
behavior doesn't change for refresh query handler.
also according to https://github.com/opensearch-project/sql/blob/main/docs/user/admin/settings.rst#pluginsqueryexecutionenginesparkrefresh_joblimit
it's intended to affect refresh job only

Reply:
> The new create index job doesn't need this?

DDL shouldn't be affected by the concurrent limit. https://github.com/opensearch-project/sql/blob/3ab785179db593a3999efffb347d897ea8364853/spark/src/test/java/org/opensearch/sql/spark/asyncquery/IndexQuerySpecTest.java#L848